src/components/atoms/ChatMessage/ChatMessage.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import classNames from "classnames";
+import dayjs from "dayjs";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrash } from "@fortawesome/free-solid-svg-icons";
+
+import { MessageToDisplay } from "types/chat";
+
+import { UserAvatar } from "components/atoms/UserAvatar";
+
+import "./ChatMessage.scss";
+
+export interface ChatProps {
+  message: MessageToDisplay;
+  onAuthorClick: () => void;
+  deleteMessage: () => void;
+}
+
+export const ChatMessage: React.FC<ChatProps> = ({
+  message,
+  onAuthorClick,
+  deleteMessage,
+}) => {
+  const { text, ts_utc, isMine, author, canBeDeleted } = message;
+
+  const timestamp = ts_utc.toMillis();
+
+  const containerStyles = classNames("chat-message", {
+    "chat-message--me": isMine,
+  });
+
+  return (
+    <div className={containerStyles}>
+      <div className="chat-message__text">{text}</div>
+      <div className="chat-message__info">
+        <UserAvatar onClick={onAuthorClick} avatarSrc={author.pictureUrl} />
+        <span onClick={onAuthorClick} className="chat-message__author">
+          {author.partyName}
+        </span>
```

Comment:
> Can we nest the UserAvatar inside a the <span>? 

Why?

> Is it a good idea to be setting onClick on a span like this?

Why not?

> Perhaps UserAvatar should have an optional prop that allows it to display the user's name as well?

It can have that, but I'm not sure it is a good idea

Reply:
I think my thinking as to group them together in the same 'container'.

---

I think my thinking on not setting onClick on span was a misremembering of some stuff from my old work's design system + the reasoning within it. I think I was remembering this bit, which is irrelevant here:

- https://bugcrowd.design/forms/index.html#example-rendered-buttons--link

And the 'rules' seem more to apply to links with onClick rather than spans/etc:

- https://bugcrowd.design/forms/index.html#buttons

---

Why don't you think it would be a good idea?